### DISCUSSION TITLE
fast way to calculate distance between points

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4335] [Q&A] [answered]
I have a use case where I want to filter a table based on a distance to a particular point, I know DuckDB don't support GIS functions, but was hoping someone has maybe find another workaround

something like this

```
SELECT *
FROM your_table
WHERE ST_Distance(location, ST_MakePoint(your_lon,your_lat)) <= 5 
```

**Comment by Alex-Monahan** [comment, marked as answer]
So, if you are looking at small distances, maybe you are comfortable assuming the earth is flat and having some errors near the poles? :-)

If so, you could use our super fast IE Join (I think @hawkfish?) with something like this:
```sql
Select * 
from my_table
Join my_location
    On my_table.lat > my_location.lat - 5/69
    And my_table.lat < my_location.lat + 5/69
    And my_table.long > my_location.long - 5/55
    And my_table.long < my_location.long + 5/55
```
Ref for lat long to miles (not accurate for longitude really): https://www.usgs.gov/faqs/how-much-distance-does-degree-minute-and-second-cover-your-maps#:~:text=One%2Ddegree%20of%20longitude%20equals,one%20second%20equals%2080%20feet.

You could likely get a bit more precise in a few ways though...haha.

**Comment by Alex-Monahan** [comment]
Or you could create these functions as macros:
https://stackoverflow.com/questions/61135374/postgresql-calculate-distance-between-two-points-without-using-postgis
